warehouse/warehouse.go

```diff
@@ -457,56 +562,60 @@ func (wh *HandleT) createLoadFiles(job *ProcessStagingFilesJobT) (err error) {
-	tableToBucketFolderMap := map[string]string{}
-	var tableToBucketFolderMapLock sync.RWMutex
-	rruntime.Go(func() {
-		func() {
-			for _, stagingFile := range job.List {
-				wh.createLoadFilesQ <- LoadFileJobT{
-					Upload:                     job.Upload,
-					StagingFile:                stagingFile,
-					Schema:                     job.Upload.Schema,
-					Warehouse:                  job.Warehouse,
-					Wg:                         wg,
-					LoadFileIDsChan:            ch,
-					TableToBucketFolderMap:     tableToBucketFolderMap,
-					TableToBucketFolderMapLock: &tableToBucketFolderMapLock,
-				}
-			}
-		}()
-	})
+	var messages []pgnotifier.MessageT
+	for _, stagingFile := range job.List {
+		payload := PayloadT{
+			UploadID:            job.Upload.ID,
+			StagingFileID:       stagingFile.ID,
+			StagingFileLocation: stagingFile.Location,
+			Schema:              job.Upload.Schema,
+			SourceID:            job.Warehouse.Source.ID,
+			DestinationID:       job.Warehouse.Destination.ID,
+			DestinationType:     wh.destType,
+			DestinationConfig:   job.Warehouse.Destination.Config,
+		}
+
+		payloadJSON, err := json.Marshal(payload)
+		if err != nil {
+			panic(err)
+		}
+		message := pgnotifier.MessageT{
+			Payload: payloadJSON,
+		}
+		messages = append(messages, message)
+	}
 
+	logger.Infof("WH: Publishing %d staging files to PgNotifier", len(messages))
 	var loadFileIDs []int64
-	waitChan := make(chan error)
-	rruntime.Go(func() {
-		func() {
-			err = wg.Wait()
-			waitChan <- err
-		}()
-	})
-	count := 0
-waitForLoadFiles:
-	for {
-		select {
-		case ids := <-ch:
-			loadFileIDs = append(loadFileIDs, ids...)
-			count++
-			logger.Debugf("WH: Processed %v staging files in batch of %v for %s:%s", count, len(job.List), wh.destType, job.Warehouse.Destination.ID)
-			logger.Debugf("WH: Received load files with ids: %v for %s:%s", loadFileIDs, wh.destType, job.Warehouse.Destination.ID)
-			if count == len(job.List) {
-				break waitForLoadFiles
-			}
-		case err = <-waitChan:
-			if err != nil {
-				logger.Errorf("WH: Discontinuing processing of staging files for %s:%s due to error: %v", wh.destType, job.Warehouse.Destination.ID, err)
-				break waitForLoadFiles
-			}
+	ch, err := wh.notifier.Publish("process_staging_file", messages)
+	if err != nil {
+		panic(err)
+	}
+
+	responses := <-ch
+	logger.Infof("WH: Received responses from PgNotifier")
+	for _, resp := range responses {
+		// TODO: make it aborted
+		if resp.Status == "aborted" {
+			continue
+		}
+		var payload map[string]interface{}
+		err = json.Unmarshal(resp.Payload, &payload)
+		if err != nil {
+			panic(err)
+		}
+		respIDs, ok := payload["LoadFileIDs"].([]interface{})
+		if !ok {
+			panic(err)
 		}
+		ids := make([]int64, len(respIDs))
+		for i := range respIDs {
+			ids[i] = int64(respIDs[i].(float64))
+		}
+		loadFileIDs = append(loadFileIDs, ids...)
 	}
-	close(ch)
+
 	timer.End()
 	if err != nil {
```

Comment:
Nitpicking: Do we need sort in 629?

Reply:
it's to get start and end loadFileID. Did you mean something else @SumanthPuram